tez-api/src/main/java/org/apache/tez/dag/api/TezConfiguration.java

```diff
@@ -2297,12 +2297,14 @@ static Set<String> getPropertySet() {
   public static final String TEZ_MRREADER_CONFIG_UPDATE_PROPERTIES = "tez.mrreader.config.update.properties";
 
   /**
-   *  Frequency at which thread dump should be captured. Supports TimeUnits.
+   *  Frequency at which thread dump should be captured. Supports TimeUnits. This is effective only
+   *  when org.apache.tez.dag.app.ThreadDumpDAGHook is configured to tez.am.hooks or
+   *  org.apache.tez.runtime.task.ThreadDumpTaskAttemptHook is configured to tez.task.attempt.hooks.
    */
   @ConfigurationScope(Scope.DAG)
   @ConfigurationProperty
   public static final String TEZ_THREAD_DUMP_INTERVAL = "tez.thread.dump.interval";
-  public static final String TEZ_THREAD_DUMP_INTERVAL_DEFAULT = "0ms";
+  public static final String TEZ_THREAD_DUMP_INTERVAL_DEFAULT = "100ms";
```

Comment:
If we introduce pluggable hooks, I think we can change the default value. We may remove NOOP_TEZ_THREAD_DUMP_HELPER, too.

Reply:
makes sense
I think for package clarity's sake, all the hook related configs can go a namespace that implies they're hooks:
```
tez.hook.thread.dump.internal
```
also:
```
TEZ_HOOK_THREAD_DUMP_INTERVAL
```

Reply:
I think you can remove the NoopTezThreadDumpHelper

Reply:
Done.
- https://github.com/apache/tez/pull/324/commits/1673bd5a88f94b097ca3c8cb627553e30efb6b6c
- https://github.com/apache/tez/pull/324/commits/79fe8d036a3fe7cae0a66163a7a8d983ffa3a14e